agent/agentpb/subscribe.proto

```diff
@@ -0,0 +1,185 @@
+    // server MAY NOT filter the results. In general it may not be possible for
+    // the server to correctly apply filters without keeping a history of every
+    // object sent on the subscription which would be prohibitively costly. So
+    // in general servers DO NOT filter streaming updates and rely on the client
+    // to apply the filter before returning to the user. The server MAY use this
+    // as a hint when shipping the initial snapshot of the result to reduce data
+    // sent unnecessarily over the wire. Currently for health endpoint the
+    // server ignores this entirely but we may change that later.
+    string Filter = 5;
+
+    // Datacenter specifies the Consul datacenter the request is targetted at.
+    // If it's not the local DC (or empty) the server will forward the stream to
+    // the remote DC and proxy the results back.
+    string Datacenter = 6;
+}
+
+// Event describes a streaming update on a subscription. Events are used both to
+// describe the current "snapshot" of the result as well as ongoing mutations to
+// that snapshot.
+message Event {
+    // Topic the event was published to
+    Topic Topic = 1;
+
+    // Key is the logical identifier for the entity that was mutated.
+    string Key = 2;
+
+    // Index is the raft index at which the mutation took place. At the top
+    // level of a subscription there will always be at most one Event per index.
+    // If multiple events are published to the same topic in a single raft
+    // transaction then the batch of events will be encoded inside a single
+    // top-level event to ensure they are delivered atomically to clients.
+    uint64 Index = 3;
+
+    // Payload is the actual event content.
+    oneof Payload {
+        // ServiceHealth is used for ServiceHealth and ServiceHealthConnect
+        // topics.
+        ServiceHealthUpdate ServiceHealth = 5;
+
+        // ACLToken represents a mutation of an ACLToken.
+        ACLTokenUpdate ACLToken = 6;
+
+        // ACLPolicy represents a mutation of an ACLPolicy.
+        ACLPolicyUpdate ACLPolicy = 7;
+
+        // ACLRole represents a mutation of an ACLRole.
+        ACLRoleUpdate ACLRole = 8;
+
+        // EndOfSnapshot indicates the event stream for the initial snapshot has
+        // ended. Subsequent Events delivered will be mutations to that result.
+        bool EndOfSnapshot = 9;
+
+        // ResumeStream indicates that the client is still up-to-date, so there
+        // will be no snapshot, and the rest of the stream will be individual
+        // update events. It disambiguates between "up to date, no snapshot" and
+        // "snapshot containes zero events".
+        bool ResumeStream = 10;
+
+        // ResetStream indicates the client should re-start the Subscribe call.
+        // It is used when the server can no longer correctly maintain the
+        // stream, for example because the ACL permisions for the token changed
```

Comment:
```suggestion
        // stream, for example because the ACL permissions for the token changed
```

Reply:
btw. figured out how to enable my spell check for .proto file ;)